lefthk/src/config.rs

```diff
@@ -42,101 +42,100 @@ macro_rules! get_keys {
+                return Err(LeftError::NumberOfKeysDiffersFromValues);
             }
-        };
-        let keybinds = command_key_pairs
-            .iter()
-            .map(|(c, k)| core_keybind {
-                command: c.clone(),
-                modifier: kb.modifier.clone(),
-                key: k.to_owned(),
-            })
-            .collect();
-        Ok(keybinds)
-    }
+            values
+                .iter()
+                .enumerate()
+                .map(|(i, v)| (core_command::Execute(v.to_owned()), keys[i].clone()))
+                .collect()
+        }
+        Command::Executes(_) => return Err(LeftError::ValuesNotFound),
+        Command::ExitChord => {
+            let keys = get_key!(kb.key);
+            vec![(core_command::ExitChord, keys)]
+        }
+        Command::Reload => {
+            let keys = get_key!(kb.key);
+            vec![(core_command::Reload, keys)]
+        }
+        Command::Kill => {
+            let keys = get_key!(kb.key);
+            vec![(core_command::Kill, keys)]
+        }
+    };
+    let keybinds = command_key_pairs
+        .iter()
+        .map(|(c, k)| core_keybind {
+            command: c.clone(),
+            modifier: kb
+                .modifier
+                .clone()
+                .unwrap_or_else(|| default_modifier.clone()),
+            key: k.to_owned(),
+        })
+        .collect();
+    Ok(keybinds)
 }
 
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 pub struct Config {
-    keybinds: Vec<Keybind>,
+    pub default_modifier: Vec<String>,
+    pub keybinds: Vec<Keybind>,
 }
 
 impl lefthk_core::config::Config for Config {
     fn mapped_bindings(&self) -> Vec<lefthk_core::config::Keybind> {
         self.keybinds
             .iter()
-            .filter_map(|kb| match TryFrom::try_from(kb.clone()) {
-                Ok(keybinds) => Some::<Vec<lefthk_core::config::Keybind>>(keybinds),
-                Err(err) => {
-                    tracing::error!("Invalid key binding: {}\n{:?}", err, kb);
-                    None
-                }
-            })
+            .filter_map(
+                |kb| match try_from(kb.clone(), self.default_modifier.clone()) {
+                    Ok(keybinds) => Some::<Vec<lefthk_core::config::Keybind>>(keybinds),
+                    Err(err) => {
+                        tracing::error!("Invalid key binding: {}\n{:?}", err, kb);
+                        None
+                    }
+                },
+            )
             .flatten()
             .collect()
     }
 }
 
-pub fn load() -> Result<Config> {
-    let path = BaseDirectories::with_prefix(lefthk_core::LEFTHK_DIR_NAME)?;
-    fs::create_dir_all(&path.get_config_home())?;
-    let file_name = path.place_config_file("config.ron")?;
-    if Path::new(&file_name).exists() {
-        let contents = fs::read_to_string(file_name)?;
+impl Config {
+    pub fn from_string(contents: String) -> Result<Self> {
```

Comment:
wouldn't it be better to make use of the `TryFrom` trait?

Reply:
This suggestion pleases me, thank you very much, I have already solved it

Reply:
:+1: